inc/structured-data-blocks/class-how-to-block.php

```diff
@@ -35,7 +35,7 @@ public function register_hooks() {
 	 * @return string The block preceded by its JSON-LD script.
 	 */
 	public function render( $attributes, $content ) {
-		if ( ! is_array( $attributes ) ) {
+		if ( ! is_array( $attributes ) || ! is_singular() ) {
```

Comment:
I would do the `is_singular` is singular check in `register_hooks`. Just to prevent calling of the `register_block_type` and maybe all actions that is called within that routine